spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -33,6 +33,13 @@
 logger = logging.getLogger(__name__)
 
 
+def validate_interpreter(pyexec):
+    """Check python excutable path and name."""
```

Comment:
Please change this to:

    Check that pyexec is a different interpreter from sys.executable